packages/react-components/src/components/VideoGallery/LocalScreenShare.tsx

```diff
@@ -1,60 +1,57 @@
-import { useTheme } from '../../theming';
-import { VideoGalleryLocalParticipant } from '../../types';
+import { CreateVideoStreamViewResult, VideoGalleryLocalParticipant, VideoStreamOptions } from '../../types';
 import { VideoTile } from '../VideoTile';
-import {
-  screenSharingContainerStyle,
-  screenSharingNotificationContainerStyle,
-  screenSharingNotificationIconContainerStyle,
-  screenSharingNotificationIconStyle,
-  screenSharingNotificationTextStyle
-} from './styles/LocalScreenShare.styles';
+import { StreamMedia } from '../StreamMedia';
+import { LoadingSpinner } from './RemoteScreenShare';
+import { _formatString } from '@internal/acs-ui-common';
 
 /**
  * A memoized version of local screen share component. React.memo is used for a performance
  * boost by memoizing the same rendered component to avoid rerendering this when the parent component rerenders.
  * https://reactjs.org/docs/react-api.html#reactmemo
  */
-export const LocalScreenShare = React.memo((props: { localParticipant: VideoGalleryLocalParticipant }) => {
-  const { localParticipant } = props;
+export const LocalScreenShare = React.memo(
+  (props: {
+    localParticipant: VideoGalleryLocalParticipant;
+    renderElement?: HTMLElement;
+    isReceiving?: boolean;
+    onCreateLocalStreamView?: (options?: VideoStreamOptions) => Promise<void | CreateVideoStreamViewResult>;
+  }) => {
+    const { localParticipant, renderElement, isReceiving, onCreateLocalStreamView } = props;
+    const locale = useLocale();
+    if (!renderElement) {
+      onCreateLocalStreamView && onCreateLocalStreamView();
+    }
 
-  const theme = useTheme();
-  const locale = useLocale();
+    if (!localParticipant || !localParticipant.isScreenSharingOn) {
+      return null;
+    }
 
-  if (!localParticipant || !localParticipant.isScreenSharingOn) {
-    return null;
-  }
-
-  const localScreenSharingNotification = (
-    <Stack horizontalAlign="center" verticalAlign="center" className={screenSharingContainerStyle}>
-      <Stack
-        horizontalAlign="center"
-        verticalAlign="center"
-        className={screenSharingNotificationContainerStyle(theme)}
-        tokens={{ childrenGap: '1rem' }}
-      >
-        <Stack horizontal verticalAlign="center" className={screenSharingNotificationIconContainerStyle}>
-          <Icon iconName="ControlButtonScreenShareStart" className={screenSharingNotificationIconStyle(theme)} />
-        </Stack>
-        <Text className={screenSharingNotificationTextStyle} aria-live="polite">
-          {locale.strings.videoGallery.screenIsBeingSharedMessage}
-        </Text>
-      </Stack>
-    </Stack>
-  );
+    const displayName = !localParticipant?.displayName
+      ? locale.strings.videoGallery.displayNamePlaceholder
+      : localParticipant?.displayName;
 
-  const displayName = !localParticipant?.displayName
-    ? locale.strings.videoGallery.displayNamePlaceholder
-    : localParticipant?.displayName;
+    const loadingMessage = displayName
```

Comment:
Should this message be `Loading your screen share`?

Reply:
Updated. Also updated the code to include disposing screen share stream. Do you mind taking a look again?